spec/integration/provider/package_spec.rb

```diff
@@ -12,7 +12,7 @@
         options = {:name => "nosuch#{provider.name}", :provider => provider.name}
 
         pkg = Puppet::Type.newpackage(options)
-        expect { pkg.provider.install }.to raise_error
+        expect { pkg.provider.install }.to raise_error(/source ?(parameter)?|gem|pip|lock file/)
```

Comment:
@joshcooper I'll have to check again tomorrow, but it returns different error messages *and* exception types in a few instances. I'll make a list of where each of these message fragments comes from.